autoload/go/highlight_test.vim

```diff
@@ -104,21 +104,45 @@ function! Test_numeric_literal_highlight() abort
   let tests = {
         \ 'lone zero': {'group': 'goDecimalInt', 'value': '0'},
         \ 'integer': {'group': 'goDecimalInt', 'value': '1234567890'},
+        \ 'integerGrouped': {'group': 'goDecimalInt', 'value': '1_234_567_890'},
+        \ 'integerErrorLeadingUnderscore': {'group': 'goDecimalErrorInt', 'value': '_1234_567_890'},
+        \ 'integerErrorTrailingUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234_567890_'},
+        \ 'integerErrorDoubleUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234__567_890'},
+        \ 'integerErrorDoubleTrailingUnderscore': {'group': 'goDecimalErrorInt', 'value': '1_234__567_890__'},
         \ 'hexadecimal': {'group': 'goHexadecimalInt', 'value': '0x0123456789abdef'},
+        \ 'hexadecimalGrouped': {'group': 'goHexadecimalInt', 'value': '0x012_345_678_9ab_def'},
         \ 'hexadecimalErrorLeading': {'group': 'goHexadecimalError', 'value': '0xg0123456789abdef'},
         \ 'hexadecimalErrorTrailing': {'group': 'goHexadecimalError', 'value': '0x0123456789abdefg'},
+        \ 'hexadecimalErrorDoubleUnderscore': {'group': 'goHexadecimalError', 'value': '0x__0123456789abdef'},
+        \ 'hexadecimalErrorDoubleTrailingUnderscore': {'group': 'goHexadecimalError', 'value': '0x__0123456789abdef__'},
```

Comment:
this test doesn't add value over the `hexadecimalErrorDoubleUnderscore`, because this test also has the double underscores in the middle of the number.